src/Compilers/Core/Portable/Syntax/SyntaxToken.cs

```diff
@@ -634,9 +634,12 @@ internal SyntaxToken GetPreviousToken(Func<SyntaxToken, bool> predicate, Func<Sy
         /// </summary>
         public Location GetLocation()
         {
-            return Node != null
-                ? this.SyntaxTree.GetLocation(this.Span)
-                : Location.None;
+            if (this.Node == null || this.SyntaxTree == null)
+            {
+                return Location.None;
+            }
+
+            return this.SyntaxTree.GetLocation(this.Span);
```

Comment:
>this.SyntaxTree [](start = 19, length = 15)

Consider avoiding duplicate evaluation of this property by caching result of the first evaluation in a local.